### PR TITLE
chore: turbo lint

### PR DESCRIPTION
**Pull request by aljo242**
Linter rules have been expanded a lot to ensure a more uniform codebase: 

```yaml
enable:
    # Check for pass []any as any in variadic func(...any).
      # Rare case but saved me from debugging a few times.
    - asasalint
    # I prefer plane ASCII identifiers.
    # Symbol `∆` instead of `delta` looks cool but no thanks.
    - asciicheck
    # Checks for dangerous unicode character sequences.
    # Super rare but why not to be a bit paranoid?
    - bidichk
    - bodyclose
    # Check whether the function uses a non-inherited context.
    - contextcheck
    # Check for two durations multiplied together.
    - durationcheck
    - depguard
    - dogsled
    - errcheck
    # Checks `Err-` prefix for var and `-Error` suffix for error type.
    - errname
    # Suggests to use `%w` for error-wrapping.
    - errorlint
    # Checks for pointers to enclosing loop variables.
    - exportloopref
    - goconst
    - gocritic
    # Forces to put `.` at the end of the comment. Code is poetry.
    - godot
    # Might not be that important, but I prefer to keep all of them.
    # `gofumpt` is amazing, kudos to Daniel Marti https://github.com/mvdan/gofumpt
    - gofmt
    - gofumpt
    - goimports
    - gosec
    - gosimple
    - govet
    - ineffassign
    - misspell
    - nakedret
    # Finds sending HTTP request without context.Context.
    - noctx
    # Finds slices that could potentially be pre-allocated.
    # Small performance win + cleaner code.
    - prealloc
    # Finds shadowing of Go's predeclared identifiers.
    # I hear a lot of complaints from junior developers.
    # But after some time they find it very useful.
    - predeclared
    # Lint your Prometheus metrics name.
    - promlinter
    # Checks that package variables are not reassigned.
    # Super rare case but can catch bad things (like `io.EOF = nil`)
    - reassign
    - revive
    - staticcheck
    - stylecheck
    # Checks that package variables are not reassigned.
    # Super rare case but can catch bad things (like `io.EOF = nil`)
    - reassign
    - typecheck
    # Test-related checks. All of them are good.
    - tenv
    - testableexamples
    - thelper
    - tparallel
    - unconvert
    - unused
    - unparam
    # Detect the possibility to use variables/constants from stdlib.
    - usestdlibvars
    # Finds wasted assignment statements.
    - wastedassign
```

Also linting tests to ensure better quality.

NO functionality is changed here.  Most of the changes are just adding periods to comments

### CODE REVIEW

File: osmosis-types/gamm/pool-models/stableswap/stableswap_pool.pb.go [hunk 1]
```diff
@@ -575,7 +575,7 @@ func (m *Pool) Unmarshal(dAtA []byte) error {
 			iNdEx = postIndex
 		case 2:
 			if wireType != 0 {
-				return fmt.Errorf("proto: wrong wireType = %d for field Id", wireType)
```
Comment (ajansari95) [inline osmosis-types/gamm/pool-models/stableswap/stableswap_pool.pb.go:578]: should not touch generated files.

File: osmosis-types/gamm/pool-models/balancer/balancerPool.pb.go [hunk 2]
```diff
@@ -1226,7 +1226,7 @@ func (m *Pool) Unmarshal(dAtA []byte) error {
 			iNdEx = postIndex
 		case 2:
 			if wireType != 0 {
-				return fmt.Errorf("proto: wrong wireType = %d for field Id", wireType)
```
Comment (ajansari95) [inline osmosis-types/gamm/pool-models/balancer/balancerPool.pb.go:1229]: generated code

File: test/e2e/configurer/chain/commands.go [hunk 3]
```diff
@@ -194,7 +195,7 @@ func (n *NodeConfig) SubmitUpgradeProposal(upgradeVersion string, upgradeHeight
 
 func (n *NodeConfig) SubmitTextProposal(text string, initialDeposit sdk.Coin) {
 	n.LogActionF("submitting text gov proposal")
-	cmd := []string{"quicksilverd", "tx", "gov", "submit-proposal", "--type=text", fmt.Sprintf("--title=\"%s\"", text), "--description=\"test text proposal\"", "--from=val", fmt.Sprintf("--deposit=%s", initialDeposit)}
+	cmd := []string{"quicksilverd", "tx", "gov", "submit-proposal", "--type=text", fmt.Sprintf("--title=\"%q\"", text), "--description=\"test text proposal\"", "--from=val", fmt.Sprintf("--deposit=%s", initialDeposit)}
```
Comment (ajansari95) [inline test/e2e/configurer/chain/commands.go:198]: why "%q" any specific reason?

File: x/airdrop/keeper/zonedrop.go [hunk 4]
```diff
@@ -98,17 +98,17 @@ func (k Keeper) AllActiveZoneDrops(ctx sdk.Context) []types.ZoneDrop {
...
 //
 // isActive:  s <= timenow <= s+D+d
-// notActive: timenow < s || timenow > s+D+d
+// notActive: timenow < s || timenow > s+D+d.
 func (k Keeper) IsActiveZoneDrop(ctx sdk.Context, zd types.ZoneDrop) bool {
 	bt := ctx.BlockTime()
 
 	// negated checks here ensure inclusive range
-	return !bt.Before(zd.StartTime) && !bt.After(zd.StartTime.Add(zd.Duration).Add(zd.Decay))
+	return bt.After(zd.StartTime) && !bt.After(zd.StartTime.Add(zd.Duration).Add(zd.Decay))
```
Comment (ajansari95) [inline x/airdrop/keeper/zonedrop.go:106]: this wont be an issue but its not inclusive of bt == Startime - just pointing the conditions are not same

File: x/airdrop/types/airdrop.pb.go [hunk 5]
```diff
@@ -647,7 +647,7 @@ func (m *ZoneDrop) Unmarshal(dAtA []byte) error {
 		switch fieldNum {
 		case 1:
 			if wireType != 2 {
-				return fmt.Errorf("proto: wrong wireType = %d for field ChainId", wireType)
```
Comment (ajansari95) [inline x/airdrop/types/airdrop.pb.go:650]: generated file

File: x/airdrop/types/messages.pb.go [hunk 6]
```diff
@@ -400,7 +400,7 @@ func (m *MsgClaim) Unmarshal(dAtA []byte) error {
 		switch fieldNum {
 		case 1:
 			if wireType != 2 {
-				return fmt.Errorf("proto: wrong wireType = %d for field ChainId", wireType)
```
Comment (ajansari95) [inline x/airdrop/types/messages.pb.go:403]: generated file

File: x/airdrop/types/query.pb.go [hunk 7]
```diff
@@ -1644,7 +1644,7 @@ func (m *QueryZoneDropRequest) Unmarshal(dAtA []byte) error {
 		switch fieldNum {
 		case 1:
 			if wireType != 2 {
-				return fmt.Errorf("proto: wrong wireType = %d for field ChainId", wireType)
```
Comment (ajansari95) [inline x/airdrop/types/query.pb.go:1647]: generated file

File: x/claimsmanager/types/claimsmanager.pb.go [hunk 8]
```diff
@@ -561,7 +561,7 @@ func (m *Claim) Unmarshal(dAtA []byte) error {
 			iNdEx = postIndex
 		case 2:
 			if wireType != 2 {
-				return fmt.Errorf("proto: wrong wireType = %d for field ChainId", wireType)
```
Comment (ajansari95) [inline x/claimsmanager/types/claimsmanager.pb.go:564]: generated file

File: x/claimsmanager/types/query.pb.go [hunk 9]
```diff
@@ -575,7 +575,7 @@ func (m *QueryClaimsRequest) Unmarshal(dAtA []byte) error {
 		switch fieldNum {
 		case 1:
 			if wireType != 2 {
-				return fmt.Errorf("proto: wrong wireType = %d for field ChainId", wireType)
```
Comment (ajansari95) [inline x/claimsmanager/types/query.pb.go:578]: generated file

**Comment by aljo242** [comment]
Will regenerate all of the proto - idk how that got in

File: x/airdrop/keeper/zonedrop.go (hunk 4 above)
Comment (aljo242) [reply x/airdrop/keeper/zonedrop.go]: reverted

File: test/e2e/configurer/chain/commands.go (hunk 3 above)
Comment (aljo242) [reply test/e2e/configurer/chain/commands.go]: we can add the quotes to the string by default instead of escaping them

**Review by ajansari95 — COMMENTED** [review]
looks good but- should revert the changes to generated files and some other comments

File: x/claimsmanager/keeper/claims.go [hunk 10]
```diff
@@ -7,12 +7,12 @@ import (
 	"github.com/ingenuity-build/quicksilver/x/claimsmanager/types"
 )
 
-func (k Keeper) NewClaim(address string, chainID string, module types.ClaimType, srcChainID string, amount uint64) types.Claim {
+func (k Keeper) NewClaim(address, chainID string, module types.ClaimType, srcChainID string, amount uint64) types.Claim {
```
Comment (joe-bowman) [inline x/claimsmanager/keeper/claims.go:10]: I hate this :/ it's far less readable IMO :(

File: x/interchainquery/keeper/msg_server.go [hunk 11]
```diff
@@ -54,7 +55,7 @@ func (k msgServer) SubmitQueryResponse(goCtx context.Context, msg *types.MsgSubm
 	noDelete := false
 	// execute registered callbacks.
 
-	keys := []string{}
+	keys := make([]string, len(k.callbacks))
```
Comment (joe-bowman) [inline x/interchainquery/keeper/msg_server.go]: `make` a slice of string x long, and then appending x strings, will give you a slice of 2x length, with x empty elements, iirc.

Comment (joe-bowman) [reply x/interchainquery/keeper/msg_server.go]: https://go.dev/play/p/aUOx9vQmUr_m?v=goprev

Comment (joe-bowman) [reply x/interchainquery/keeper/msg_server.go]: Will likely panic on line 67 for referencing a non-existent element

**Review by joe-bowman — APPROVED** [review]
Lgtm now

**Comment by joe-bowman** [comment]
@ajansari95

**Review by ajansari95 — APPROVED** [review]
lgtm

**Pull request merged**